kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -99,6 +179,20 @@ impl<'tcx> KaniAttributes<'tcx> {
                     ),
                 );
             }
+            if kind.is_function_contract_api() && !queries.function_contracts_enabled() {
```

Comment:
I have mixed feelings about failing the compilation if a function has a contract annotation and contracts is disabled.

I believe we should only fail if the contract is being used for verification (i.e.: either there's a `[kani::proof_for_contracts]` or `[kani::stub_verified]`.

Reply:
We talked about this today and if I recall we decided this would be an error, right?

Reply:
This should also mean it automatically only applies if the contracts are reachable I think.